src/Frontend/components/portal/CreateAccount.js

```diff
@@ -10,24 +10,71 @@
 
 'use strict';
 
-import React     from 'react';
-import Translate from 'react-translate-component';
-import Form      from './signup/Form';
-import InfoBox   from './signup/InfoBox';
+import React, { Component, PropTypes }     from 'react';
+import Translate                           from 'react-translate-component';
+import { connect            }              from 'react-redux';
+import { bindActionCreators }              from 'redux';
+import Form                                from './signup/Form';
+import InfoBox                             from './signup/InfoBox';
+import * as userActions                    from '../../actions/userActions';
 
 /**
- * Markup component which renders the `create account` page.
+ * Presentational component for the sign-up page
  *
- * @returns {React.Element} The markup.
+ * @author Benjamin Bieler <ben@benbieler.com>
  */
-export default () => {
-  return (
-    <div>
-      <h1><Translate content="pages.portal.head" /></h1>
+class CreateAccount extends Component {
+
+  static PropTypes = {
+    actions: PropTypes.object.isRequired
+  };
+
+  static contextTypes = {
+    store: PropTypes.object
+  };
+
+  /**
+   * Handle the onSubmit event
+   *
+   * @param data
+   * @param recaptchaHash
+   */
+  handleSubmit = (data) => {
+    this.props.actions.sign_up.createAccount(data);
+  };
+
+  render() {
+
+    const { store }      = this.context;
+    let name_suggestions = [], success = false;
+
+    store.subscribe(() => {
+      name_suggestions = store.getState().user.registration.name_suggestions;
+      success          = store.getState().user.registration.success;
+    });
+
+    return (
       <div>
-        <InfoBox />
-        <Form />
+        <h1><Translate content="pages.portal.head" /></h1>
+        <div>
+          <InfoBox />
+          <Form onSubmit={this.handleSubmit} name_suggestions={name_suggestions} success={success} />
+        </div>
       </div>
-    </div>
-  );
-};
+    );
+  }
+}
+
+const mapStateToProps = state => ({});
```

Comment:
why not just putting the state definition from the store subscription in here? We could even avoid this `store.subscribe` call.